src/line-top-index.js

```diff
@@ -69,53 +69,62 @@ export default class LineTopIndex {
     let oldEnd = traverse(start, oldExtent)
     let newEnd = traverse(start, newExtent)
 
+    let isInsertion = isZero(oldExtent)
     let startNode = this.iterator.insertNode(start)
-    let endNode = this.iterator.insertNode(oldEnd)
+    let endNode = this.iterator.insertNode(oldEnd, !isInsertion)
 
     startNode.priority = -1
     this.bubbleNodeUp(startNode)
     endNode.priority = -2
     this.bubbleNodeUp(endNode)
 
-    if (startNode !== endNode) {
-      let blockIdsToMove = new Set
+    startNode.blockIds.forEach((id) => {
+      if (!this.inclusiveBlockIds.has(id)) return
 
-      startNode.blockIds.forEach((id) => {
-        if (this.inclusiveBlockIds.has(id)) {
-          startNode.blockIds.delete(id)
-          blockIdsToMove.add(id)
-        }
-      })
+      startNode.blockIds.delete(id)
+      startNode.blockHeight -= this.blockHeightsById[id]
+      startNode.distanceFromLeftAncestor.pixels -= this.blockHeightsById[id]
 
-      if (startNode.right) {
-        this.collectBlockIdsForSubtree(startNode.right, blockIdsToMove)
-        startNode.right = null
-      }
+      endNode.blockIds.add(id)
+      endNode.blockHeight += this.blockHeightsById[id]
```

Comment:
Yeah, another reason why I let this tiny bit of code to be duplicated is that I realized that we overlooked decrementing the height because we thought the two code paths were identical, whereas when we move stuff from `startNode` we need to perform some additional operations. Not sure if it's exactly accidental duplication, but I wasn't even sure the two code paths needed to evolve simultaneously.